File: app/src/main/java/com/transax/hahiye/viewModel/LoginViewModel.java
package com.transax.hahiye.viewModel;

import android.arch.lifecycle.ViewModel;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.View;

import com.transax.hahiye.callback.ResultCallback;
import com.transax.hahiye.model.UserModel;

public class LoginViewModel extends ViewModel {

    private ResultCallback resultCallback;
    private UserModel userModel = new UserModel();

    LoginViewModel(ResultCallback resultCallback) {
        this.resultCallback = resultCallback;
    }

    public TextWatcher getEmailFromView() {
        return new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {

            }

            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {

            }

            @Override
            public void afterTextChanged(Editable s) {
                userModel.setEmail(s.toString());
            }
        };
    }

    public TextWatcher getPasswordFromView() {
        return new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {

            }

            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {

            }

            @Override
            public void afterTextChanged(Editable s) {
                userModel.setPassword(s.toString());
            }
        };
    }

    public void performLogin(View view) {
        if (userModel.isValidData())
            resultCallback.onSuccess("Information are correct");
        else
            resultCallback.onError("Provide correct information");
    }
}

File: app/src/main/java/com/transax/hahiye/viewmodel/LoginViewModel.kt
package com.transax.hahiye.viewmodel

import android.app.Activity
import android.arch.lifecycle.ViewModel
import android.text.Editable
import android.text.TextWatcher
import android.view.View
import android.widget.Toast
import com.transax.hahiye.model.UserModel
import com.transax.hahiye.view.callback.LoginResultCallback

class LoginViewModel(private val callback: LoginResultCallback) : ViewModel() {

    val userModel: UserModel = UserModel()

    fun getEmailFromView(): TextWatcher {
        return object : TextWatcher {
            override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {

            }

            override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {

            }

            override fun afterTextChanged(s: Editable) {
                userModel.email = s.toString()
            }
        }
    }

    fun getPasswordFromView(): TextWatcher {
        return object : TextWatcher {
            override fun afterTextChanged(s: Editable?) {

            }

            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {

            }

            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
                userModel.password = s.toString()
            }

        }
    }

    fun performLogin(view :View) {
        if (userModel.isValidData())
            callback.onSuccess("Information are correct")
        else
            callback.onError("Provide correct information")
    }
}